src/books/repositories/typeorm/book-repository-typeorm.adapter.ts

```diff
@@ -1,40 +1,132 @@
+import { Model } from 'mongoose';
+import { InjectModel } from '@nestjs/mongoose';
+import { BookMongo, BookMongoDocument } from '../../schemas/book.schema';
 
 @Injectable()
-export class BookRepositoryTypeOrmAdapter extends BooksRepository {
+export class BookRepositoryTypeOrmAdapter
+  extends BooksRepository
+  implements OnApplicationBootstrap {
   constructor(
     @InjectRepository(Book)
     private booksRepository: Repository<Book>,
+    @InjectModel(BookMongo.name) private bookModel: Model<BookMongoDocument>,
   ) {
     super();
   }
 
+  async onApplicationBootstrap() {
+    console.log('OnApplicationBootstrap from BookRepositoryTypeOrmAdapter : Sync mongo with mysql');
+    const allBooks: Book[] = await this.booksRepository.find();
+    await this.bookModel.deleteMany({});
+    for (let index = 0; index < allBooks.length; index++) {
+      const element: Book = allBooks[index];
+      const createdBook: BookMongoDocument = new this.bookModel(element);
+      await createdBook.save();
+    }
+  }
+
   async findOneById(id: string): Promise<Book> {
-    return this.booksRepository.findOne(id);
+    // return this.booksRepository.findOne(id);
+    const document: BookMongoDocument = await this.bookModel.findOne({ id: id }).exec();
+    if (document === null) {
+      return undefined;
+    } else {
+      return new Book(document.toObject());
+    }
   }
 
   async findAll(): Promise<Book[]> {
-    return this.booksRepository.find();
+    // return this.booksRepository.find();
+
+    // Je crois que je me suis perdu, help pour reecrire avec async/await
+    // return new Promise((resolve, reject) => {
+    //   this.bookModel
+    //     .find({})
+    //     .exec()
+    //     .then((value: BookMongoDocument[]) => {
+    //       const tmp: Book[] = [];
+    //       for (let index = 0; index < value.length; index++) {
+    //         tmp[index] = new Book(value[index].toObject());
+    //       }
+    //       resolve(tmp);
+    //     })
+    //     .catch((error) => {
+    //       reject(error);
+    //     });
+    // });
+
+    const documents: BookMongoDocument[] = await this.bookModel.find({}).exec();
+    const output: Book[] = [];
+    for (let index = 0; index < documents.length; index++) {
+      output[index] = new Book(documents[index].toObject());
+    }
+    return output;
+  }
+
+  private async handleCreateMongo(id: string): Promise<void> {
+    const toWrite: Book = await this.booksRepository.findOne(id);
+    const createdBook: BookMongoDocument = new this.bookModel(toWrite);
+    //? await or not for last ? or return ? or return await ?
+    await createdBook.save();
   }
 
   async create(id: string, payload: CreateBookDto): Promise<Book> {
-    return this.booksRepository.save({
+    const output: Promise<Book> = this.booksRepository.save({
       id,
       ...payload,
     });
+    //? Possible de reecrire avec async ???
+    output.then(async () => {
+      await this.handleCreateMongo(id);
+    });
```

Comment:
Voila une version transformant tes promises en async/await 🙂 
```suggestion
    const output = await this.booksRepository.save({
      id,
      ...payload,
    });
    await this.handleCreateMongo(id);
```

Reply:
je pense pas je je peux pas expliquer par texte si on pouvait se parler ce serait plus simple